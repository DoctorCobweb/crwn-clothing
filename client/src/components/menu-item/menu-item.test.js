import React from 'react';
import { shallow } from 'enzyme';

import { MenuItem } from './menu-item.component';

describe('MenuItem component', () => {
  let wrapper;
  let mockProps;
  let mockPush;

  beforeEach(() => {
    mockPush = jest.fn();
    mockProps = {
      title: 'mockTitle',
      imageUrl: 'https://example.com/image1',
      size: 'mockSize',
      linkUrl: 'mockLinkUrl',
      match: {
        url: 'https://exmaple.com/',
      },
      history: {
        push: mockPush,
      },
    };

    wrapper = shallow(<MenuItem {...mockProps} />);
  });

  it('should render component', () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it('should correctly call history.push() when MenuItemContainer is clicked', () => {
    wrapper.find('MenuItemContainer').simulate('click');
    expect(mockPush).toHaveBeenCalled();
    expect(mockPush).toHaveBeenCalledWith(`${mockProps.match.url}${mockProps.linkUrl}`);
  });

  it('should have correctly set imageUrl prop on BackgroundImageContainer', () => {
    expect(wrapper.find('BackgroundImageContainer').prop('imageUrl')).toBe(mockProps.imageUrl);
  });

  it('should pass size to MenuItemContainer as the prop size', () => {
    expect(wrapper.find('MenuItemContainer').prop('size')).toBe(mockProps.size);
  });

  it('should correctly set ContentTitle text', () => {
    expect(wrapper.find('ContentTitle').text()).toBe(mockProps.title.toUpperCase());
  });

  it('should correct set ContentSubtitle text', () => {
    expect(wrapper.find('ContentSubtitle').text()).toBe('SHOP NOW');
  });
});