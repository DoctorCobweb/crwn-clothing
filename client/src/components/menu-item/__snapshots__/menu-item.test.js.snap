// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuItem component should render component 1`] = `
"<MenuItemContainer size=\\"mockSize\\" onClick={[Function: onClick]}>
  <BackgroundImageContainer className=\\"background-image\\" imageUrl=\\"https://example.com/image1\\" />
  <ContentContainer className=\\"content\\">
    <ContentTitle>
      MOCKTITLE
    </ContentTitle>
    <ContentSubtitle>
      SHOP NOW
    </ContentSubtitle>
  </ContentContainer>
</MenuItemContainer>"
`;
